larpandoracontent/LArHelpers/LArFormattingHelper.cc

```diff
@@ -0,0 +1,184 @@
+
+    void LArFormattingHelper::Reset(std::ostream &stream)
+    {
+        LArFormattingHelper::ResetColour(stream);
+        LArFormattingHelper::ResetStyle(stream);
+    }
+    
+    //--------------------------------------------------------------------------------------------------------------------------------------
+
+    void LArFormattingHelper::PrintFormatCharacter(const unsigned int &code, std::ostream &stream)
+    {
+        stream << LArFormattingHelper::GetFormatCharacter(code);
+    }
+    
+    //--------------------------------------------------------------------------------------------------------------------------------------
+
+    std::string LArFormattingHelper::GetFormatCharacter(const unsigned int &code)
+    {
+        return ("\x1B[" + std::to_string(code) + "m");
+    }
+    
+    //--------------------------------------------------------------------------------------------------------------------------------------
+
+    void LArFormattingHelper::PrintHeader(const std::string &title, const unsigned int &width)
+    {
+        LArFormattingHelper::SetStyle(BOLD);
+        std::cout << std::endl << std::string(width, '-') << "\r-" << title << std::endl;
+        LArFormattingHelper::Reset();
+    }
+    
+    //--------------------------------------------------------------------------------------------------------------------------------------
+
+    void LArFormattingHelper::PrintRule(const unsigned int &width)
+    {
+        LArFormattingHelper::PrintHeader("", width);
+    }
+    
+    //--------------------------------------------------------------------------------------------------------------------------------------
+    //--------------------------------------------------------------------------------------------------------------------------------------
+    
+    LArFormattingHelper::Table::Table(const std::vector<std::string> &columnTitles, const unsigned int &precision) :
+        m_columnTitles(columnTitles),
+        m_precision(precision)
+    {
+        m_stringstream.precision(m_precision);
+
+        for (const std::string &title : m_columnTitles)
+            m_widths.push_back(title.length());
+    }
+    
+    //--------------------------------------------------------------------------------------------------------------------------------------
+    
+    void LArFormattingHelper::Table::CheckForSeparatorColumn()
+    {
+        if (m_columnTitles[m_elements.size() % m_columnTitles.size()] == "")
+        {
+            m_format.push_back("");
+            m_elements.push_back("");
+        }
+    }
+    
+    //--------------------------------------------------------------------------------------------------------------------------------------
+    
+    bool LArFormattingHelper::Table::IsSeparatorColumn(const unsigned int &column)
+    {
+        if (column>=m_columnTitles.size())
+            throw StatusCodeException(STATUS_CODE_OUT_OF_RANGE);
+
+        return (m_columnTitles[column] == "");
```

Comment:
Again non const, but name makes it sound like it should be a const member function.